src/arche/readers/items.py

```diff
@@ -9,7 +9,46 @@
 from scrapinghub.client.jobs import Job
 
 
-class Items(ABC):
+class Items:
+    def __init__(self, df: pd.DataFrame, expand: bool = False):
+        self._df = self.process_df(df)
+        self._flat_df = None
+        self.expand = expand
+
+    @property
+    def df(self):
```

Comment:
If `df` will be accessible for users, why not using `self.df` instead of `self._df` and get rid of this property?

Reply:
When I did this I was thinking that it will help to not modify `df`. Alas, it this doesn't help (but `df.copy()` does)